src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from 'react';
+import { useMatch } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+
+  const match = useMatch('/people/:personSlug');
+  const selectedPersonSlug = match?.params.personSlug;
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then((result) => {
+        setIsLoading(false);
+        setPeople(result.map(person => ({
+          ...person,
+          mother: result
+            .find(mother => mother.name === person.motherName),
```

Comment:
```suggestion
             const getParent = (name: string | null) => (
    people.find((parent) => parent.name === name) || null
  );
```
since we use the same code more than once I would recommend you to implement a helper function so you can use it in both cases, without typing twice.